validator/client/propose_protect.go

```diff
@@ -33,17 +35,19 @@ func (v *validator) preBlockSignValidations(ctx context.Context, pubKey [48]byte
 		)
 	}
 
-	_, exists, err = v.db.ProposalHistoryForSlot(ctx, pubKey, block.Slot)
+	prevSigningRoot, exists, err := v.db.ProposalHistoryForSlot(ctx, pubKey, block.Slot)
 	if err != nil {
 		if v.emitAccountMetrics {
 			ValidatorProposeFailVec.WithLabelValues(fmtKey).Inc()
 		}
 		return errors.Wrap(err, "failed to get proposal history")
 	}
-	// If a proposal exists in our history for the slot, we assume it is slashable.
-	// TODO(#7848): Add a more sophisticated strategy where if we indeed have the signing root,
-	// only blocks that have a conflicting signing root with a historical proposal are slashable.
-	if exists {
+
+	// If a proposal exists in our history for the slot, we check the following:
+	// If the signing root is nil, then we consider it slashable. If signing root is not nil,
+	// we check if it is different than the incoming block's signing root. If that is the case,
+	// we consider that proposal slashable.
+	if exists && (prevSigningRoot == params.BeaconConfig().ZeroHash || prevSigningRoot != signingRoot) {
```

Comment:
Shouldn't we stop a proposal if a block proposal was already submitted? 
Why allow another block to be proposed, even if it has exactly the same information? 

Whatever benefit there is, I dont know if it's worth the risk of proposing a block when the validator has already proposed one.

Reply:
This is totally allowed. The only slashable offense is if the signing root does not match but the slot matches

Reply:
Also this is a test case for the EIP spec tests. We expect a block with same slot and same signing root to go through ok